servers/quarkus-server/src/main/java/org/projectnessie/server/authz/CompiledAuthorizationRules.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) 2020 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.server.authz;
+
+import io.quarkus.runtime.Startup;
+import java.util.HashMap;
+import java.util.Map;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import org.projectnessie.cel.tools.Script;
+import org.projectnessie.cel.tools.ScriptException;
+import org.projectnessie.server.config.QuarkusNessieAuthorizationConfig;
+import org.projectnessie.services.cel.CELUtil;
+
+/**
+ * Compiles the authorization rules from {@link QuarkusNessieAuthorizationConfig} at startup and
+ * provides access to them via {@link CompiledAuthorizationRules#getRules()}.
+ */
+@Singleton
+@Startup
+public class CompiledAuthorizationRules {
+  private final QuarkusNessieAuthorizationConfig config;
+  private final Map<String, Script> compiledRules;
```

Comment:
can you copy this into an immutable map so we aren't returning a mutable map in teh getter

Reply:
thanks for bringing this up, this is definitely needed. I pushed a fix.